config/plugins/interactive_environments/jupyter/static/js/jupyter.js

```diff
@@ -0,0 +1,85 @@
+function message_failed_auth(password){
+    toastr.info(
+        "Automatic authorization failed. You can manually login with:<br>" + password + "<br> <a href='https://github.com/bgruening/galaxy-ipython/wiki/Automatic-Authorization-Failed' target='_blank'>More details ...</a>",
+        "Please login manually",
+        {'closeButton': true, 'timeOut': 100000, 'tapToDismiss': false}
+    );
+}
+
+function message_failed_connection(){
+    toastr.error(
+        "Could not connect to Jupyter Notebook. Please contact your administrator. <a href='https://github.com/bgruening/galaxy-ipython/wiki/Could-not-connect-to-IPython-Notebook' target='_blank'>More details ...</a>",
+    "Security warning",
+        {'closeButton': true, 'timeOut': 20000, 'tapToDismiss': true}
+    );
+}
+
+function message_no_auth(){
+    // No longer a security issue, proxy validates Galaxy session token.
+    /*
+    toastr.warning(
+        "IPython Notebook was lunched without authentication. This is a security issue. <a href='https://github.com/bgruening/galaxy-ipython/wiki/IPython-Notebook-was-lunched-without-authentication' target='_blank'>More details ...</a>",
+        "Security warning",
+        {'closeButton': true, 'timeOut': 20000, 'tapToDismiss': false}
+    );
+    */
+}
+
+
+/**
+ * Load an interactive environment (IE) from a remote URL
+ * @param {String} password: password used to authenticate to the remote resource
+ * @param {String} notebook_login_url: URL that should be POSTed to for login
+ * @param {String} notebook_access_url: the URL embeded in the page and loaded
+ *
+ */
+function load_notebook(password, notebook_login_url, notebook_access_url){
+    $( document ).ready(function() {
+        // Test notebook_login_url for accessibility, executing the login+load function whenever
+        // we've successfully connected to the IE.
+        test_ie_availability(notebook_login_url, function(){
+            _handle_notebook_loading(password, notebook_login_url, notebook_access_url);
+        });
+    });
+}
+
+/**
+ * Must be implemented by IEs
+ */
+function _handle_notebook_loading(password, notebook_login_url, notebook_access_url){
+    if ( ie_password_auth ) {
+        // Make an AJAX POST
+        $.ajax({
```

Comment:
Yeah, I left it in intentionally. If someone cares about this extra layer of security it can be used.
Do you want me to remove it? You could potentially and listen for open ports on your Docker host and enter a Jupyter session easily (if it's not firewall protected).